connector/spanmetricsconnector/connector.go

```diff
@@ -54,6 +54,8 @@ type connectorImp struct {
 
 	resourceMetrics *cache.Cache[resourceKey, *resourceMetrics]
 
+	resourceMetricsKeyAttributes map[string]bool
```

Comment:
For the sake of memory usage optimisations, you'd want to use `map[string]struct{}` here since the empty struct allocates no memory and you can use the second returned parameter to check if it exists within a set.